src/server.ts

```diff
@@ -354,18 +353,7 @@ export class SubscriptionServer {
                   params.variables,
                   params.operationName);
 
-                if (!isAsyncIterable(promiseOrIterable) && isPromise(promiseOrIterable)) {
-                  executionIterable = promiseOrIterable;
-                } else if (isAsyncIterable(promiseOrIterable)) {
-                  executionIterable = Promise.resolve(promiseOrIterable as any as AsyncIterator<ExecutionResult>);
-                } else {
-                  // Unexpected return value from execute - log it as error and trigger an error to client side
-                  console.error('Invalid `execute` return type! Only Promise or AsyncIterable are valid values!');
-
-                  this.sendError(connectionContext, opId, {
-                    message: 'GraphQL execute engine is not available',
-                  });
-                }
+                executionIterable = Promise.resolve(promiseOrIterable);
```

Comment:
hmm..
since we now should support:
`Result`
`Promise<Result>`
`Promise<AsyncIterator<Result>>`
we should rename `promiseOrIterable` to `promiseOrResult`
also, we should `Promise.resolve` only if `!isPromise(..)`
and `executionIterable` to `executionPromise`

Also,
the createAsyncIterable in line 366 is probably not needed anymore =)

Reply:
As `Promise.resolve` handles the case when passed in a Promise, I'm inclined to just use that. I've pushed an updated that always makes execution a Promise which I think makes the handling cleaner.